portal/components/table/index.tsx

```diff
@@ -0,0 +1,178 @@
+'use client'
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useWindowSize } from 'hooks/useWindowSize'
+import { useRef } from 'react'
+import Skeleton from 'react-loading-skeleton'
+
+import { useInfiniteScroll } from './_hooks/useInfiniteScroll'
+import { useScrollbarDetection } from './_hooks/useScrollbarDetection'
+import { useTableData } from './_hooks/useTableData'
+import { Column, ColumnHeader } from './table'
+
+export type TableProps<T> = {
+  columns: ColumnDef<T, unknown>[]
+  data?: T[]
+  getRowKey?: (row: T, index: number) => string
+  hasMore?: boolean
+  loading?: boolean
+  loadingMore?: boolean
+  onLoadMore?: VoidFunction
+  priorityColumnIdsOnSmall?: string[]
+  skeletonRows?: number
+  smallBreakpoint?: number
+}
+
+export function Table<T>({
```

Comment:
It's a little unclear to me what T stands for here.

Perhaps we should name it

`TDRowData`, similar to how `@tanstack/table` names it.


```suggestion
export function Table<TRowData>({
```

And share this name into the props too. What do you think?